src/openseadragon.js

```diff
@@ -816,6 +816,32 @@ window.OpenSeadragon = window.OpenSeadragon || function( options ){
                     canvasElement.getContext( '2d' ) );
     }());
 
+    /**
+     * @private
+     * @inner
+     * @function
+     * @returns {Number} The device's pixel density ratio, or null if canvas isn't supported by the browser.
+     */
+    $.pixelDensityRatio = (function () {
+        if ( $.supportsCanvas ) {
```

Comment:
Is it not already doing this? I basically followed the closure style from $.supportsCanvas, which as far as I can tell it calls the function on first run to generate a value and then assigns it to $.supportsCanvas. If I misread, though, happy to change it.